http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -397,6 +422,18 @@ private HttpRequest buildJettyRequest(Request finalRequest)
         jettyRequest.timeout(requestTimeoutMillis, MILLISECONDS);
         jettyRequest.idleTimeout(idleTimeoutMillis, MILLISECONDS);
 
+        // client authentications
+        if (jettyRequest.getURI().getScheme().equalsIgnoreCase("https")) {
+            jettyRequest.header(io.airlift.security.utils.SecurityUtil.REALM_IN_CHALLENGE, "true");
+            AuthenticationStore authStore = httpClient.getAuthenticationStore();
+            if (securityConfig != null && securityConfig.enabled()) {
+                String authScheme = securityConfig.getAuthScheme().toString();
+                if (authStore.findAuthentication(authScheme, jettyRequest.getURI(), null) == null) {
+                        httpClient.getAuthenticationStore().addAuthentication(
```

Comment:
Is this going to add an authentication to the http client for each request? When do they get cleaned up?